ts/features/idpay/onboarding/xstate/machine.ts

```diff
@@ -285,8 +322,61 @@ const createIDPayOnboardingMachine = () =>
                 target: "DISPLAYING_INITIATIVE"
               }
             ]
+          },
+          onDone: {
+            target: "ACCEPTING_REQUIRED_CRITERIA"
+            // triggered by the 'final' substate
+          },
+          states: {
+            EVALUATING_SELF_CRITERIA: {
+              tags: [LOADING_TAG],
+              always: [
+                {
+                  target: "DISPLAYING_BOOL_CRITERIA",
+                  cond: "hasBoolRequiredCriteria"
+                },
+                {
+                  target: "CYCLING_MULTI_CRITERIA",
+                  cond: "hasMultiRequiredCriteria"
+                }
+              ]
+            },
+            DISPLAYING_BOOL_CRITERIA: {
+              tags: [WAITING_USER_INPUT_TAG],
+              entry: "navigateToBoolSelfDeclarationsScreen",
+              on: {
+                ACCEPT_REQUIRED_BOOL_CRITERIA: [
+                  {
+                    target: "CYCLING_MULTI_CRITERIA",
+                    cond: "hasMultiRequiredCriteria"
+                  },
+                  {
+                    target: "MULTI_PREREQUISITES_CYCLE_OVER"
+                  }
+                ]
+              }
+            },
+            CYCLING_MULTI_CRITERIA: {
+              tags: [WAITING_USER_INPUT_TAG],
+              entry: "navigateToMultiSelfDeclarationsScreen",
+              on: {
+                ADD_MULTI_CONSENT: {
+                  actions: "addMultiConsent"
+                },
+                ALL_CRITERIA_ACCEPTED: {
+                  target: "MULTI_PREREQUISITES_CYCLE_OVER"
+                },
+                GO_BACK: {
+                  target: "DISPLAYING_BOOL_CRITERIA"
+                }
```

Comment:
Here we are going back to the bool criteria without checking if it's present. 

```suggestion
                GO_BACK: [
                  {
                    target: "DISPLAYING_BOOL_CRITERIA",
                    cond: "hasBoolRequiredCriteria"
                  },
                  {
                    target: "#IDPAY_ONBOARDING.DISPLAYING_INITIATIVE"
                  }
                ]
```